org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/DocumentLifeCycleHandlerTest.java

```diff
@@ -886,6 +886,80 @@ public void testDocumentMonitorClosedDocument() throws Exception {
 		documentMonitorAfterClose.checkChanged(); // Version not changed (null -> null)
 	}
 
+	@Test
+	public void testFileRenameForTypeDeclaration() throws Exception {
+		Mockito.lenient().when(clientPreferences.isWorkspaceApplyEditSupported()).thenReturn(true);
+		IJavaProject javaProject = newEmptyProject();
+		IPackageFragmentRoot sourceFolder = javaProject.getPackageFragmentRoot(javaProject.getProject().getFolder("src"));
+		IPackageFragment pack1 = sourceFolder.createPackageFragment("test1", false, null);
+
+		String content = "package test1;\n"
+				+ "\n"
+				+ "class Foo {\n"
+				+ "}";
+
+		ICompilationUnit cu1 = pack1.createCompilationUnit("Other.java", content, false, null);
+
+		openDocument(cu1, cu1.getSource(), 1);
+		saveDocument(cu1);
+
+		assertEquals(0, getClientRequests("applyEdit").size());
```

Comment:
Maybe I could add a comment explaining that `applyEdit` is what we use to check the rename is going to happen.

Reply:
I agree.